src/models/team.ts

```diff
@@ -0,0 +1,249 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import type * as BalenaSdk from '..';
+import type { InjectedDependenciesParam, PinePostResult, Team } from '..';
+import * as errors from 'balena-errors';
+import { isId, isNotFoundResponse, mergePineOptions } from '../util';
+
+const getTeamModel = function (deps: InjectedDependenciesParam) {
+	const { pine, sdkInstance } = deps;
+
+	/**
+	 * @summary Creates a new Team
+	 * @name create
+	 * @public
+	 * @function
+	 * @memberof balena.models.team
+	 *
+	 * @description This method creates a new team.
+	 *
+	 * @param {Number} organizationSlugOrId - Required: the organization slug or id the team will be part of.
+	 * @param {String} name - Required: the name of the team that will be created.
+	 *
+	 * @fulfil {Object} - Team
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.create(1239948, 'MyTeam').then(function(team) {
+	 * 	console.log(team);
+	 * });
+	 *
+	 * @example
+	 * balena.models.team.create('myOrgHandle', 'MyTeam')
+	 * .then(function(team) {
+	 *   console.log(team);
+	 * });
+	 */
+	const create = async function (
+		organizationSlugOrId: string | number,
+		name: string,
+	): Promise<PinePostResult<Team>> {
+		const orgId = isId(organizationSlugOrId)
+			? organizationSlugOrId
+			: (await sdkInstance.models.organization.get(organizationSlugOrId))?.id;
+
+		if (orgId == null) {
+			throw new errors.BalenaOrganizationNotFound(organizationSlugOrId);
+		}
+
+		const teams = await sdkInstance.models.team.getAllByOrganization(orgId);
+
+		if (teams.some((t) => t.name === name)) {
+			throw new Error(
+				`A team with this name already exists in the organization. Organization: ${orgId}, Name: ${name}`,
+			);
+		}
+
+		return pine.post({
+			resource: 'team',
+			body: {
+				name,
+				belongs_to__organization: orgId,
+			},
+		});
+	};
+
+	/**
+	 * @summary Get all Teams of a specific Organization
+	 * @name getAllByOrganization
+	 * @public
+	 * @function
+	 * @memberof balena.models.team
+	 *
+	 * @param {Number} organizationSlugOrId - Required: the organization slug or id the team is part of.
+	 * @param {Object} [options={}] - extra pine options to use
+	 *
+	 * @fulfil {Object[]} - Teams
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.team.getAllByOrganization(123).then(function(teams) {
+	 * 	console.log(teams);
+	 * });
+	 *
+	 * @example
+	 * balena.models.team.getAllByOrganization('MyOrganizationHandle').then(function(teams) {
+	 * 	console.log(teams);
+	 * });
+	 */
+	const getAllByOrganization = function (
+		organizationSlugOrId: string | number,
+		options: BalenaSdk.PineOptions<BalenaSdk.Team> = {},
+	): Promise<BalenaSdk.Team[]> {
+		return pine.get({
```

Comment:
* Mark it as async
* then add a throw in case organizationSlugOrId is unexpected

Reply:
do you mean by adding a pine.get(resource: 'organization', ...) or by checking if the argument is of type string | num ber ?

Reply:
I'm fine with either :)  IE
* Checking for number/string and throwing invalid parameter error
* doing just a `sdk.modelsorganization.get()`